Drivers/MotorChannel/Src/ZP_D_PWMChannel.cpp

```diff
@@ -0,0 +1,42 @@
+#include "ZP_D_PWMChannel.hpp"
+
+PWMChannel::PWMChannel(uint16_t pin_num, GPIO_TypeDef* port, TIM_HandleTypeDef* timer,
+                        uint16_t timer_channel, uint32_t clock_frequency) :    pin_num_(pin_num),
+                                                    port_(port),
+                                                    timer_(timer),
+                                                    timer_channel_(timer_channel),clock_frequency_(clock_frequency) {}
+
+void PWMChannel::setup(void) {
+	/*Sets up the PWM, prescaler, duty cycle ranges,
+	 * and starts the timer*/
+	period_ticks_ = timer_->Init.Period;
+	min_signal = period_ticks_ * 0.05; // sets counts for 5% duty cycle
+	max_signal = period_ticks_ * 0.10; // sets counts for 10% duty cycle
+
+	//Calculate new prescaler
+	uint16_t prescaler_ = (clock_frequency_ / DESIRED_FREQUENCY
+			/ period_ticks_) - 1;
+
+	__HAL_TIM_SET_PRESCALER(timer_, prescaler_);
+	HAL_TIM_PWM_Start(timer_, timer_channel_);
+
+
+	isSetup = true;
+}
```

Comment:
Any reason to have this "setup" function instead of just doing this work in the constructor?

Reply:
No, I can add it in there.